### PR TITLE
[DOCS] Update docstrings for some of actions.py

### PR DESCRIPTION
**Pull request by talagluck**
Document part of actions.py

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: great_expectations/checkpoint/actions.py [hunk 1]
```diff
@@ -143,18 +142,31 @@ class SlackNotificationAction(ValidationAction):
...
+            {
+               "module_name": "great_expectations.render.renderer.slack_renderer",
+               "class_name": "SlackRenderer",
+           }
+        slack_webhook (str): Optional. The incoming Slack webhook to which to send notification.
+        slack_token (str): Optional. Token from Slack app. Used when not using slack_webhook.
+        slack_channel (str): Optional. Slack channel to receive notification. Used when not using slack_webhook.
+        notify_on (str): Specifies validation status that triggers notification. One of "all", "failure", "success".
+        notify_with (list[str]): Optional list of DataDocs site names to display  in Slack messages. Defaults to all.
+        show_failed_expectations (bool): Shows a list of failed expectation types.
```
Comment (anthonyburdi) [inline great_expectations/checkpoint/actions.py]: Oooh - bonus points if you can add these type annotations instead to the `__init__` method signature.

File: great_expectations/checkpoint/actions.py [hunk 2]
```diff
@@ -168,20 +180,6 @@ def __init__(
         notify_with=None,
         show_failed_expectations=False,
     ) -> None:
```
Comment (anthonyburdi) [inline great_expectations/checkpoint/actions.py:183]: Let's keep some docstring on this and other `__init__` methods, even if just "Create SlackNotificationAction". See https://google.github.io/styleguide/pyguide.html#384-classes for more info about class docstrings.

**Review by anthonyburdi — COMMENTED** [review]
Just a couple small suggested tweaks (can be in a followup PR or I'll make them if I have some time). I'm not approving also because the preview deploy is failing and I'm trying to debug why.

**Comment by talagluck** [comment]
Thanks, @anthonyburdi! Updates made. Looks like the build is failing, but I'm not sure why.

**Pull request merged**